workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/RegistryValidator.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.validator.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.validator.Validator;
+import io.hyscale.controller.manager.RegistryManager;
+import io.hyscale.controller.model.WorkflowContext;
+import io.hyscale.controller.util.RegistryAndDockerValidatorUtil;
+
+@Component
+public class RegistryValidator implements Validator<WorkflowContext> {
+	private static final Logger logger = LoggerFactory.getLogger(ClusterValidator.class);
+
+	@Autowired
+	private RegistryManager registryManager;
+
+	@Override
+	public boolean validate(WorkflowContext context) throws HyscaleException {
+		logger.debug("Starting K8s cluster validation");
+		if(!RegistryAndDockerValidatorUtil.isValidate(context.getServiceSpec())) {
+			return false;
+		}
+		return registryManager.getImageRegistry("") != null ? true : false;
```

Comment:
This will always return null, we need to fetch the registry name from service spec and pass that.
To improve efficiency should we also store a list of validated registry to avoid rechecking as number of times single registry will be used across service specs.